static/app/views/dashboards/widgets/timeSeriesWidget/releaseBubbles/utils/createReleaseBuckets.spec.tsx

```diff
@@ -0,0 +1,160 @@
+import {createReleaseBuckets} from 'sentry/views/dashboards/widgets/timeSeriesWidget/releaseBubbles/utils/createReleaseBuckets';
+
+function createTimeSeries(numItems: number, intervalMs = 1000, start = new Date()) {
+  const timeSeries = [
+    {
+      data: Array.from(Array(numItems)).map((_, i) => ({
+        value: 0, // don't care about value
+        timestamp: new Date(start.getTime() + i * intervalMs).toISOString(),
+      })),
+    },
+  ];
+
+  // find min/max timestamp of *all* timeSeries
+  let minTime: Date | undefined;
+  let maxTime: Date | undefined;
+
+  for (const currentSeries of timeSeries) {
+    if (currentSeries.data.length < 2) {
+      continue;
+    }
+
+    const firstData = currentSeries.data[0];
+    const lastData = currentSeries.data[currentSeries.data.length - 1];
+    // I hope `data` is sorted
+    if (!minTime || new Date(firstData!.timestamp) < minTime) {
+      minTime = new Date(firstData!.timestamp);
+    }
+    if (!maxTime || new Date(lastData!.timestamp) > maxTime) {
+      maxTime = new Date(lastData!.timestamp);
+    }
+  }
+
+  return {minTime: minTime?.getTime(), maxTime: maxTime?.getTime()};
+}
+
+describe('createReleaseBuckets', () => {
+  it.each([
+    [0, 0],
+    [1, 0],
+    [2, 10],
+    [10, 10],
+    [20, 10],
+    [50, 10],
+    [100, 10],
+    [300, 10],
+  ])(
```

Comment:
A test to check for creating different bucket counts would be good here, I think, since the count is configurable

Reply:
[`7fee5b5` (#85270)](https://github.com/getsentry/sentry/pull/85270/commits/7fee5b54e3a27c21076cbb73123cac1f33447bbd)